ext/curl/interface.c

```diff
@@ -1185,12 +1188,54 @@ PHP_MINIT_FUNCTION(curl)
 		return FAILURE;
 	}
 
+	zend_class_entry ce;
+	INIT_CLASS_ENTRY(ce, "Curl", curl_object_methods);
+	ce.create_object = curl_create_object;
+	ce.ce_flags |= ZEND_ACC_FINAL;
+	curl_ce = zend_register_internal_class(&ce);
+
+	memcpy(&curl_object_handlers, &std_object_handlers, sizeof(zend_object_handlers));
+	curl_object_handlers.offset = XtOffsetOf(php_curl, std);
+	curl_object_handlers.free_obj = curl_free_obj;
+	curl_object_handlers.get_gc = curl_get_gc;
+	curl_object_handlers.get_constructor = curl_get_constructor;
+
+	curl_multi_register_class();
+	curl_share_register_class();
 	curlfile_register_class();
 
 	return SUCCESS;
 }
 /* }}} */
 
+/* Curl class */
+
+static zend_object *curl_create_object(zend_class_entry *class_type) {
+	php_curl *intern = zend_object_alloc(sizeof(php_curl), class_type);
+	memset(intern, 0, sizeof(php_curl) - sizeof(zend_object));
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+	intern->std.handlers = &curl_object_handlers;
+
+	return &intern->std;
+}
+
+static zend_function *curl_get_constructor(zend_object *object) {
+	zend_throw_error(NULL, "Cannot directly construct Curl, use curl_init() instead");
+	return NULL;
+}
+
+static HashTable *curl_get_gc(zend_object *object, zval **table, int *n)
+{
+	php_curl *curl = curl_from_obj(object);
+
+	*table = &curl->postfields;
```

Comment:
postfield is a copy of the array or object that has been set via `curl_setopt(CURLOPT_POSTFIELDS, postfields)`, so yes, it may contain cyclic references.

Reply:
I got rid of conditional inclusion of `postfields`. What should I do if it contains cyclic references? Do I have to iterate the hash map, similarily to what SPL's linked list does?

Reply:
No, what you have here is sufficient.